modules/state_holder.py

```diff
@@ -0,0 +1,149 @@
+import json
+import logging
+import os
+import pathlib
+import pickle
+
+import redis
+from fastapi import Request
+from starlette.middleware.base import BaseHTTPMiddleware
+
+
+def _get_redis_client(redis_address: str = '') -> redis.Redis:
+    if not redis_address:
+        redis_address = os.getenv('REDIS_ADDRESS', '')
+
+    if redis_address:
+        redis_client = redis.Redis.from_url(url=redis_address)
+    else:
+        redis_client = None
+    return redis_client
+
+
+def _load_module(path):
+    import importlib.util
+    module_spec = importlib.util.spec_from_file_location(os.path.basename(path), path)
+    module = importlib.util.module_from_spec(module_spec)
+    module_spec.loader.exec_module(module)
+
+    return module
+
+
+class _StateSerializer:
```

Comment:
In python, when defining a class, it is always a good habbit to subclass some other class or `object`